### ISSUE TITLE
Link: No `globalPlacement()` equivalent for LinkElement

### ISSUE THREAD

**Issue by WandererFan** [FreeCAD/FreeCAD#13978]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

TechDraw and Measure workbenches require shapes with their net placement applied. For objects derived from GeoFeature, this is simple - selection provides myObject and mySubelement, Part::Feature::getShape(myObject).getSubShape(mySubElement) gets the  shape and myObject->globalPlacement() provides the net placement to apply to the shape.

When myObject is a LinkElement, or the subject of a LinkElement, we can still retrieve the shape, but there is no equivalent to globalPlacement. 

In order to obtain the net placement in cases where a LinkElement is involved, the selection result must be interpreted using the resolve method and we must traverse the dependency tree from myObject up to a top level object, gathering placements at each step, then apply these placements (in top down order) to the shape.

This tree crawling process should be hidden from application code by a general purpose getLocatedShape(mySelection) that handles all object types that provide a shape.

### Full version info

```shell
OS: Linux Mint 20.3 (X-Cinnamon/cinnamon)
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.37088 +13 (Git)
Build type: debug
Branch: CosmeticReview5
Hash: 2ecbac7416045619e0a021da9f9a47f54f67130c
Python 3.8.10, Qt 5.12.8, Coin 4.0.0, Vtk , OCC 7.7.1
Locale: English/Canada (en_CA)
Installed mods: 
  * BOLTSFC 2022.11.5
  * sheetmetal 0.2.63
  * fasteners 0.4.56
  * FreeCAD_Electric
  * freecad.gears 1.0.0
  * Curves 0.6.29
  * Assembly4 0.50.9
  * A2plus 0.4.64
```

### Subproject(s) affected?

Techdraw

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by WandererFan** [comment]
related forum discussion: https://forum.freecad.org/viewtopic.php?t=87609

**Comment by wolfgangr** [comment]
from my mentioned forum discussion (well, monologue...) I'd add:
- calculated spreadsheet locations fail to retrieve global placement of linked shapes
- same with Part attachments
- same with attached fastener WB objects

**Comment by WandererFan** [comment]
This situation also holds for Link objects.

**Comment by WandererFan** [comment]
Another forum discussion: https://forum.freecad.org/viewtopic.php?t=45406

**Comment by adrianinsaval** [comment]
duplicate of #11977?

**Comment by WandererFan** [comment]
yes, +/- a duplicate.   It would be good to "merge" the comments here with #11977 rather than losing them by just closing this issue, but I don't know how to do that except by retyping.
